csharp/src/Ice/Internal/UdpTransceiver.cs

```diff
@@ -554,8 +518,9 @@ public ConnectionInfo getInfo(bool incoming, string adapterName, string connecti
         {
             EndPoint localEndpoint = Network.getLocalAddress(_fd);
 
-            if (_state == StateNotConnected) // TODO: eliminate all these states
+            if (_state == StateNotConnected) // a server endpoint
             {
+                Debug.Assert(incoming);
```

Comment:
Question: do we create a new Connection and potentially ConnectionInfo for each incoming UDP request?

Reply:
Looking at the IncomingConnectionFactory I believe there is a single udp Connection created by the factory constructor. But then caching the info object wouldn't be correct.

Reply:
I believe we create a new connection for each request.  And udp bi-dir relies on these per-request connections:
https://github.com/zeroc-ice/ice/blob/41307f90a6d1ebe097f3aef5b5b5e802dd78f543/csharp/test/Ice/udp/AllTests.cs#L200